src/platform/magento2/order/orderProcessor.js

```diff
@@ -0,0 +1,302 @@
+import {productsEquals} from 'vsf-utilities';
+import order from '../../../api/order';
+const orderUtilities = require('./order-utilities');
+const apiUtilities = require('./api-utilities');
+const config = require('config');
+const redis = require('../../../lib/redis');
+const redisClient = redis.getClient(config)
+const logger = console;
+const countryMapper = require('../../../lib/countrymapper')
+
+const TOTAL_STEPS = 4;
+var currentStep = 1;
+
+/**
+ * @param {json} orderData order data in format as described in '../models/order.md'
+ * @returns {Promise}
+ */
+async function cartIdPrepare (orderData) {
+  const api = apiUtilities.getApi(orderData.store_code);
+  const isThisAuthOrder = orderUtilities.isThisAuthOrder(orderData);
+  const {
+    cart_id: cartId,
+    user_id: userId
+  } = orderData;
+
+  return isThisAuthOrder ? api.cart.create(null, userId) : (cartId ? new Promise((resolve, reject) => {
+    resolve(cartId)
+  }) : api.cart.create(null));
+}
+
+/**
+ * @param {json} orderData order data in format as described in '../models/order.md'
+ * @param job
+ * @param {Function} done callback - @example done(new Error()) - to acknowledge problems
+ * @returns {Promise<*>}
+ */
+async function processCart ({ orderData, job, done }) {
+  try {
+    await updateCartItems({ job, orderData });
+  } catch (err) {
+    if (job) job.attempts(3).backoff({ delay: 30 * 1000, type: 'fixed' }).save();
+    return done(orderUtilities.composeError('Error while adding products.', err));
+  }
+
+  const shippingInformation = await assignAddresses({ job, orderData, done });
```

Comment:
If it returns `done` inside - won't function still go further? I think you could miss something while rewriting :(

Reply:
hmm.. I will have to check. :) 
I might, because I wasn't using `done` in the beginning at all.. when I play around with `await..async`